cmds/kill/kill.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Kill kills processes.
+// -l lists the signal names.
+// -[s]name is the message to send. On some systems this is a string,
+// on others a number.
+// [pid] is the list of pids. It doesn't make a lot of sense to have
+// no arguments but the definition allows it.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+var (
+	l = flag.Bool("l", false, "list signal names")
+)
+
+func usage() {
+	log.Fatalf("Usage: kill -l | kill -[s |-signal ]signame pid [pid...]")
+	os.Exit(1)
+}
+
+func main() {
+	a := os.Args
+	op := a[1]
+	p := a[2:]
+	// sadly, we can not use flag. Well, we could,
+	// it would be pretty cheap to just put in every
+	// possible flag as a switch, but it just gets
+	// messy.
+	//
+	// kill is one of the old school commands (1971)
+	// and arg processing was not really standard back then.
+	// Also, note, the -l has no meaning on Plan 9 or Harvey
+	// since signals on those systems are arbitrary strings.
+
+	if op[0:2] == "-l" {
+		if len(a) > 2 {
+			usage()
+		}
+		fmt.Print(siglist())
+		os.Exit(1)
+	}
+
+	// N.B. Be careful if you want to change this. It has to continue to work if
+	// the signal is an arbitrary string. We take the path that the signal
+	// has to start with a -, or might be preceded by -s or --string.
+	if op[0] != '-' {
```

Comment:
It does not relate to that, I think it should be here as this is where we start processing signal names?